src/ch/ch_domain.c

```diff
@@ -149,6 +149,12 @@ virCHDomainObjPrivateAlloc(void *opaque)
         VIR_FREE(priv);
         return NULL;
     }
+
+    if (!(priv->devs = virChrdevAlloc())) {
```

Comment:
how/where is priv->devs used? I don't see any references to devs below? Also this doesn't correspond to number of serial/console devices configured in domain xml?

Reply:
Currently cloud-hypervisor can only handle a single serial or console device and the way libvirt itself handles virsh console is it looks only for a single device. I decided to avoid trying to replicate the allocations per device specified in the domain xml since they wouldn't be used anyway at this time.

The devs struct is used by a vir utility function to populate information about the character device and isn't used directly by the ch code.